hydra_python_core/doc_maker.py

```diff
@@ -247,12 +248,44 @@ def create_property(supported_prop: Dict[str, Any]) -> HydraClassProp:
     for k, literal in doc_keys.items():
         result[k] = input_key_check(
             supported_prop, k, "supported_prop", literal)
+    # Check if it's a link property
+    if isinstance(result["property"], Dict):
+        result["property"] = create_link_property(result["property"])
     # Create the HydraClassProp object
     prop = HydraClassProp(result["property"], result["title"], required=result["required"],
                           read=result["readable"], write=result["writeable"])
     return prop
 
 
+def create_link_property(
+        link_prop_dict: Dict[str, Any]) -> HydraLink:
+    """Create HydraLink objects for link properties in the API Documentation."""
+    id_ = link_prop_dict["@id"]
+
+    doc_keys = {
```

Comment:
I don't really understand why we are giving all these fields to create a Hydra:Link instance? A Link will have only the URI and the type defined. Why are we giving it a title, description, etc?

Reply:
>A Link will have only the URI and the type defined. Why are we giving it a title, description, etc?
No a link will have title, defination, range, domain, etc.

From issue-tracker vocab:

property |  
-- | --
@id | "vocab:Comment/issue"
@type | "hydra:Link"
label | "issue"
description | "The issue this comment belongs to"
domain | "vocab:Comment"
range | "vocab:Issue"


`hydra:Link` is already defined in the Spec and I am nowhere redefining it, we are just providing support to adding entries of predicates with type (`hydra:Link`). And such predicates will have title, domain, range, etc.

>A Link will have only the URI and the type defined

A predicate of type `hydra:Link` will have all the necessary fields and will be stored in the APIDoc, hydrus will only return a link to the instance, so yes the response body will only have a URI but the definition of that property in the APIDoc have all other fields.